app/models/labware.rb

```diff
@@ -224,7 +242,7 @@ def find_from_any_barcode(source_barcode)
         nil
       elsif /\A[0-9]{1,7}\z/.match?(source_barcode)
         # Just a number
-        joins(:barcodes).where('barcodes.barcode LIKE "__?_"', source_barcode).first # rubocop:disable Rails/FindBy
+        joins(:barcodes).order(:id).find_by('barcodes.barcode LIKE "__?_"', source_barcode)
```

Comment:
This was mostly to make rubocop happy. I added order by to ensure actual behaviour isn't changed. (Although in practice duplicate barcodes are exceedingly rare)